beetmoverscript/script.py

```diff
@@ -154,6 +156,43 @@
     copy_beets(context, candidates_keys_checksums, releases_keys_checksums)
 
 
+async def push_to_maven(context):
+    """Push artifacts to locations expected by maven clients (like mvn or gradle)"""
+    context.artifacts_to_beetmove = task.get_upstream_artifacts_with_zip_extract_param(context)
+    context.release_props = get_release_props(context)
+    context.checksums = dict()  # Needed by downstream calls
+    context.raw_balrog_manifest = dict()    # Needed by downstream calls
+
+    mapping_manifest = generate_beetmover_manifest(context)
+    validate_bucket_paths(context.bucket, mapping_manifest['s3_bucket_path'])
+
+    expected_files = maven.get_maven_expected_files_per_archive_per_task_id(
+        context.artifacts_to_beetmove, mapping_manifest
+    )
+
+    extracted_paths_per_archive = zip.check_and_extract_zip_archives(
+        context.artifacts_to_beetmove,
+        expected_files,
+        context.config.get('zip_extract_max_file_size_in_mb', ZIP_MAX_FILE_SIZE_IN_MB)
+    )
+
+    number_of_extracted_archives = len(extracted_paths_per_archive)
```

Comment:
I somehow feel like these sanity checks could be moved elsewhere in a separate function but I may be wrong. The other `push_to_*` methods largely avoided doing any checks (well, frankly there are some in`push_to_releases` IIRC). I have no strong feelings though and this isn't blocking. 

It's just that these functions, the way I see them, are blocks of logic to follow-up what's going on. But the actual logic, validations, checks and processing happens behind the scenes. What do you think?